keras_cv/bounding_box/transform.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv.bounding_box.converters import convert_format
+
+H_AXIS = -3
+W_AXIS = -2
+
+
+def transform_from_corners_fn(
+    bounding_boxes,
+    *,
+    transform_corners_fn,
+    bounding_box_format,
+    images=None,
+    clip_boxes=True,
+):
+    """Transforms bounding boxes by applying a function to each of its corners.
+
+    Args:
+      bounding_boxes: a 2D (unbatched) or 3D (batched) Tensor of
+        bounding boxes, potentially ragged.
+      transform_corners_fn: a function to apply to each of the corners
```

Comment:
this seems like a weird abstraction to offer as a package.  I'm not sure I'm comfortable exposing this API.  does this basically transform_format(), apply transform, then transform back?